docs/source/user_guide/code-snippets.md

```diff
@@ -21,62 +21,151 @@ The ability to reuse pieces of code allows users to avoid doing repetitive work,
-[code snippets metadata](https://github.com/elyra-ai/elyra/blob/master/elyra/metadata/schemas/code-snippet.json)
+```
+Available metadata instances for code-snippets (includes invalid):
 
-This is a sample json file:
-```json
-{
-	"schema_name": "code-snippet",
-	"display_name": "is_even",
-	"metadata": {
-		"description": "Check if number is even",
-		"language": "python",
-		"code": [
-			"def is_even(num):",
-			"   return num % 2 == 0"
-		],
-        "tags": [
-		  "python"
-        ]
-	}
-}
+Schema   	Instance		Resource  
+---------   	--------		--------
+code-snippet   	preview_dataframe	/.../metadata/code-snippets/preview_dataframe.json 
 ```
-NOTE: `code` field content must be a string array split by line.
 
-Once code snippet extension is in use, the application component retrieves the metadata from the server
-through a REST API, and displays each snippet by `display_name` in the extension UI.
-Each snippet item can then be expanded on click, also displaying the `code` content.
+To format the output as JSON run `elyra-metadata list code-snippets --json`. Note that the JSON export includes the content of the metadata files, not just their names.
 
-![Code Snippet Sample](../images/code-snippet-expanded.png)
+#### Creating a code snippet
 
-By clicking on the **copy** button, the content of the code snippet is copied into the system clipboard.
+To create a code snippet:
 
-The **insert** button inserts the code snippet content into the editor tab in focus. If the widget in focus is not an editor, an error is displayed.
-Code snippets are automatically added as code blocks in markdown files and notebook markdown cells.
-When inserting snippets into executable editors (ie. a notebook code cell or a python file editor), the extension will verify kernel language compatibility, warning the user when a mismatch is detected.
+```bash
+elyra-metadata install code-snippets \
+	--display_name="Preview DataFrame" \
+	--description="Preview Pandas DataFrame" \
+	--tags="['Python', 'Pandas']" \
+	--language="Python" \
+	--code="['df.head(5)']"
+```
+
+Refer to the [Code snippet properties](#code-snippet-properties) section for an explanation of the parameters.
+
+#### Modifying a code snippet
+
+To modify a code snippet:
+
+```bash
+elyra-metadata install code-snippets \
+	--replace \
+	--name="preview_dataframe" \
+	--display_name="Preview DataFrame" \
+	--description="Preview Pandas DataFrame" \
+	--tags="['Python', 'Pandas']" \
+	--language="Python" \
+	--code="['# Display first 5 rows', 'df.head(5)']"
+```
+
+Refer to the [Code snippet properties](#code-snippet-properties) section for an explanation of the parameters. Note that you must specify the `--name` parameter and the `--replace` parameter. 
+
+#### Deleting a code snippet
+
+To delete a code snippet run the following command, replacing the code snippet name as appropriate.
+
+```bash
+elyra-metadata remove code-snippet --name="preview_dataframe"
```

Comment:
This command throws the error `Subcommand 'code-snippet' is invalid.`
```suggestion
elyra-metadata remove code-snippets --name="preview_dataframe"
```

Reply:
arg, yes. thank you. I never understood why we use plural, even though most commands operate on a single instance.